core/js/src/main/scala/fs2/compression/CompressionPlatform.scala

```diff
@@ -22,6 +22,91 @@
 package fs2
 package compression
 
-private[compression] trait CompressionPlatform[F[_]]
+private[compression] trait CompressionPlatform[F[_]] {
+
+  /** Returns a pipe that incrementally compresses input into the GZIP format
+    * as defined by RFC 1952 at https://www.ietf.org/rfc/rfc1952.txt. Output is
+    * compatible with the GNU utils `gunzip` utility, as well as really anything
+    * else that understands GZIP. Note, however, that the GZIP format is not
+    * "stable" in the sense that all compressors will produce identical output
+    * given identical input. Part of the header seeding is arbitrary and chosen by
+    * the compression implementation. For this reason, the exact bytes produced
+    * by this pipe will differ in insignificant ways from the exact bytes produced
+    * by a tool like the GNU utils `gzip`.
+    *
+    * GZIP wraps a deflate stream with file attributes and stream integrity validation.
+    * Therefore, GZIP is a good choice for compressing finite, complete, readily-available,
+    * continuous or file streams. A simpler deflate stream may be better suited to
+    * real-time, intermittent, fragmented, interactive or discontinuous streams where
+    * network protocols typically provide stream integrity validation.
+    *
+    * @param bufferSize The buffer size which will be used to page data
+    *                   into chunks. This will be the chunk size of the
+    *                   output stream. You should set it to be equal to
+    *                   the size of the largest chunk in the input stream.
+    *                   Setting this to a size which is ''smaller'' than
+    *                   the chunks in the input stream will result in
+    *                   performance degradation of roughly 50-75%. Default
+    *                   size is 32 KB.
+    * @param deflateLevel     level the compression level (0-9)
+    * @param deflateStrategy  strategy compression strategy -- see `java.util.zip.Deflater` for details
+    * @param modificationTime optional file modification time
+    * @param fileName         optional file name
+    * @param comment          optional file comment
+    */
+  def gzip(
+      bufferSize: Int = 1024 * 32,
+      deflateLevel: Option[Int] = None,
+      deflateStrategy: Option[Int] = None,
+      modificationTime: Option[Nothing] = None,
+      fileName: Option[Nothing] = None,
+      comment: Option[Nothing] = None
```

Comment:
Using `Option[Nothing]` keeps us source-compatible with JVM while only accepting `None`. Furthermore, thanks to type erasure it leaves the door open to add support for these in the future if Node.js ever gets them.
- https://github.com/nodejs/node/issues/39545